storey/flow.py

```diff
@@ -532,18 +532,19 @@ def __init__(self, max_in_flight=8, **kwargs):
 
     async def _worker(self):
         try:
+            job = await self._q.get()
+            event = None
             while True:
-                job = await self._q.get()
                 if job is _termination_obj:
                     for pending_event in self._pending_by_key.values():
                         if pending_event.pending and not pending_event.in_flight:
-                            resp = await self._process_event(pending_event.pending)
+                            resp = await self._safe_process_events(pending_event.pending)
                             for event in pending_event.pending:
                                 await self._handle_completed(event, resp)
                     if self._q.empty():
                         break
                     else:
-                        await self._q.put(_termination_obj)
+                        job = _termination_obj
```

Comment:
`job` is already `termination_obj`, no?

Reply:
Hm yeah, maybe this is not needed? Why would q not be empty?

Reply:
I replaced the if/else with just a break.